### DISCUSSION TITLE
fio single job randrw md5 verify failure

### DISCUSSION THREAD

**Discussion by changchengx** [axboe/fio#1708] [Q&A] [answered]
For data integrity md5 verification, is it normal to use this option when doing randrw test?
For example: ```sudo fio --bs=1M --iodepth=256 --numjobs=1 --runtime=60 --filename=/dev/vda --rw=randrw --name=test-rw --ioengine=libaio --direct=1 --time_based --verify=md5```

**Comment by changchengx** [comment]
@sitsofe could you help check this question?

**Comment by vincentkfu** [comment]
Run this with `--debug=io,verify` to gain some insight into what fio is doing. It looks like fio is writing IOs with verification headers but just doing regular reads.

You could try something like:

```
fio --bs=1M --iodepth=256 --numjobs=1 --filename=/dev/vda--rw=randwrite --name=test-rw --ioengine=libaio --direct=1 --verify=md5 --debug=io,verify
```

Fio will randomly write the device and then verify/read all of the blocks written.

**Comment by changchengx** [comment]
@vincentkfu I'll try to run the test with ```--debug=io,verify```. Is it right to do md5 verification under **```randrw```** test instead of **```randwrite```** test?

**Comment by vincentkfu** [comment, marked as answer]
I would use the command I suggested which includes `rw=randwrite` for your test. Drop `time_based` and `runtime` and just let fio write the entire device. After the write phase is complete fio will then verify the data just written.
